src/main/java/com/google/firebase/messaging/FirebaseMessaging.java

```diff
@@ -139,6 +143,191 @@ protected String execute() throws FirebaseMessagingException {
     };
   }
 
+  /**
+   * Sends each message in the given list via Firebase Cloud Messaging.
+   * Unlike {@link #sendAll(List)}, this method makes a single HTTP call for each message in the
+   * given array.
+   *
+   * <p>The responses list obtained by calling {@link BatchResponse#getResponses()} on the return
+   * value corresponds to the order of input messages.
+   *
+   * @param messages A non-null, non-empty list containing up to 500 messages.
+   * @return A {@link BatchResponse} indicating the result of the operation.
+   * @throws FirebaseMessagingException If an error occurs while handing the messages off to FCM for
+   *     delivery. An exception here or a {@link BatchResponse} with all failures indicates a total
+   *     failure -- i.e. none of the messages in the list could be sent. Partial failures or no
```

Comment:
Suggest "that is," or "meaning that" instead of "i.e."

https://developers.google.com/style/abbreviations#dont-use